app/components/agent/Show.js

```diff
@@ -34,11 +35,22 @@ class AgentShow extends React.Component {
   };
 
   componentDidMount() {
-    this._agentRefreshInterval = setInterval(this.fetchUpdatedData, 5::seconds);
+    // Only bother setting up the delayed load and refresher if we've got an
+    // actual agent to play with.
+    if(this.props.agent && this.props.agent.id) {
+      this._agentRefreshInterval = setInterval(this.fetchUpdatedData, 5::seconds);
+
+      // Once the agent's show page has mounted in DOM, switch `isMounted` to
+      // true which will trigger a load of all the additional information we need
+      // to show this page.
+      this.props.relay.setVariables({ isMounted: true });
+    }
   }
 
   componentWillUnmount() {
-    clearInterval(this._agentRefreshInterval);
+    if(this._agentRefreshInterval) {
+      clearInterval(this._agentRefreshInterval);
+    }
```

Comment:
`clearInterval` doesn’t actually care if it’s given nothing! which is weird, I guess